test/controllers/contact_controller_test.exs

```diff
@@ -0,0 +1,36 @@
+defmodule ResearchResource.ContactControllerTest do
+  use ResearchResource.ConnCase
+
+  test "GET /contact", %{conn: conn} do
+    conn = get conn, "/contact"
+    assert html_response(conn, 200) =~ "Contact Us"
+  end
+
+  test "POST /contact/create - request call back all fields", %{conn: conn} do
+    conn = post conn, "/contact", callback: %{"name" => "Test", "phone" => "123"}
+    assert html_response(conn, 302) =~ "redirected"
+    assert redirected_to(conn) == contact_path(conn, :index)
+    assert get_flash(conn, :info)
+  end
+
+  test "POST /contact/create - call back missing fields", %{conn: conn} do
+    conn = post conn, "/contact", callback: %{"name" => "Test"}
+    assert html_response(conn, 302) =~ "redirected"
+    assert redirected_to(conn) == contact_path(conn, :index)
+    assert get_flash(conn, :error)
+  end
+
+  test "POST /contact/create - message all fields", %{conn: conn} do
+    conn = post conn, "/contact", message: %{"name" => "Test", "message" => "hello"}
+    assert html_response(conn, 302) =~ "redirected"
+    assert redirected_to(conn) == contact_path(conn, :index)
+    assert get_flash(conn, :info)
+  end
+
+  test "POST /contact/create - message missing fields", %{conn: conn} do
+    conn = post conn, "/contact", message: %{"name" => "Test"}
+    assert html_response(conn, 302) =~ "redirected"
+    assert redirected_to(conn) == contact_path(conn, :index)
+    assert get_flash(conn, :error)
+  end
```

Comment:
Looks good, I may have looked to abstract out some of this to get rid of the repetition